File: my-folder/1256-rank-transform-of-an-array/solution.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] newArr = Arrays.copyOf(arr, arr.length);
        int r = 1;
        Arrays.sort(newArr);
        HashMap<Integer, Integer> hashy = new HashMap<Integer, Integer>();
        for(int i : newArr){
            if(!hashy.containsKey(i)){
                hashy.put(i, r);
                r++;
            }
        }
        

        for(int i = 0; i < arr.length; i++){
            int rank = hashy.get(arr[i]);
            newArr[i] = rank;
        }
        return newArr;

    }
}

File: my-folder/1256-rank-transform-of-an-array/solution.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        dupe = sorted(set(arr))
        rank = {}
        for r, num in enumerate(dupe):
            rank[num] = r + 1 
        
        return [rank[i] for i in arr]

        
